dallinger/command_line/lib/ec2.py

```diff
@@ -79,21 +80,49 @@ def list_regions():
     print(pd.DataFrame(region_metadata).to_markdown())
 
 
-def get_instances(region_name=None):
+def get_instance_details(instance_types, region_name=None):
+    response = requests.get(
+        "https://ec2.shop",
+        params={
+            "filter": ",".join(instance_types),
+            "region": region_name,
+        },
+        headers={
+            "accept": "json",
+        },
+    )
+    if response.status_code != 200:
+        print(f"Failed to get details for {instance_types} in {region_name}")
+        return None
+
+    price_df = pd.DataFrame(response.json()["Prices"])
+
+    if len(price_df) != len(instance_types):
+        print(f"Failed to get all details for {instance_types} in {region_name}")
+    return price_df
+
+
+def get_instances(region_name):
     reservations = get_ec2_client(region_name).describe_instances()["Reservations"]
     instances = []
     for reservation in reservations:
         for instance in reservation["Instances"]:
+            try:
+                name = instance["Tags"][0]["Value"]
+            except KeyError:
+                name = "Unnamed"
             instances.append(
                 {
-                    "name": instance["Tags"][0]["Value"],
+                    "name": name,
                     "instance_id": instance["InstanceId"],
                     "instance_type": instance["InstanceType"],
-                    "availability_zone": instance["Placement"]["AvailabilityZone"],
+                    "region": region_name,
                     "state": instance["State"]["Name"],
                     "public_dns_name": instance["PublicDnsName"],
                     # 'public_ip_address': instance['PublicIpAddress'],
                     "pem": instance["KeyName"],
+                    # Duration since instance was started
+                    "uptime": time.time() - instance["LaunchTime"].timestamp(),
```

Comment:
Doesn't this make an assumption about what the local timezone is?

Reply:
Implemented your suggestion.